provisioning/webhook.go

```diff
@@ -77,7 +77,7 @@ func EnableValidatingWebhook(info *ProvisioningInfo, mgr manager.Manager) error
 	// we might not have a baremetalCR (when disabled), so we have no where to store
 	// the expectedGeneration, so just fake it.
 	expectedGeneration := int64(0)
-	_, _, err := resourceapply.ApplyValidatingWebhookConfiguration(info.Client.AdmissionregistrationV1(), info.EventRecorder, instance, expectedGeneration)
+	_, _, err := resourceapply.ApplyValidatingWebhookConfiguration(info.KubeClient.AdmissionregistrationV1(), info.EventRecorder, instance, expectedGeneration)
```

Comment:
ProvisioningInfo is created in here https://github.com/openshift/cluster-baremetal-operator/blob/5f4ae6841cd64cf74e9c8cd31523dc84ff7bda6f/main.go#L110 without setting KubeClient. But it accesses it in here.